3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,77 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  const data = await fetch(url);
+  const response = await data.json();
+  return response;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  try {
+    const data = await fetchData('https://pokeapi.co/api/v2/pokemon?limit=151');
+    const select = document.querySelector('select');
+    if (!select.textContent) {
```

Comment:
This works: you're not adding the pokemon twice. But, even if the pokemon are already loaded, you do do the network request a second time. Moving the request into the `if`'s body you could also avoid that, i.e.

```js
async function fetchAndPopulatePokemons() {
  try {
    const select = document.querySelector('select');
    if (!select.textContent) {
      const data = await fetchData('https://pokeapi.co/api/v2/pokemon?limit=151');
```